### PR TITLE
#6051 Fix null validInstanceStatusList when getting InstanceType

### PR DESCRIPTION
**Pull request by keytiong**
Signed-off-by: Key Tiong Tan <keytiong@hotmail.com>

# Description

Set validInstanceTypeList when creating InstanceType in OMRSRepositoryContentManager

Fixes # 6051

# How Has This Been Tested?

Unit test included

# Any additional notes for reviewers?

### CODE REVIEW

**Review by planetf1 — CHANGES_REQUESTED** [review]
I approved the code runs for you .

The maven build failed due to the file 'open-metadata-implementation/repository-services/repository-services-implementation/src/test/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentManagerTest.java' not having a valid SPDX header.

```
 /* SPDX-License-Identifier: Apache 2.0 */
 /* Copyright Contributors to the ODPi Egeria project. */
 ```
 
 The snyk errors are a process issue. See https://github.com/odpi/egeria/issues/6118 - they should be removed on your next run

**Comment by planetf1** [comment]
This build failed with a network issue
Restarting the build

File: open-metadata-implementation/repository-services/repository-services-implementation/src/test/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.UUID;
+
+import static org.mockito.Mockito.*;
+import static org.testng.Assert.assertEquals;
+
+
+public class OMRSRepositoryContentManagerTest {
+
+
+
```
Comment (mandy-chessell) [inline open-metadata-implementation/repository-services/repository-services-implementation/src/test/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentManagerTest.java:24]: Why do we need Mockito for this test?

File: open-metadata-implementation/repository-services/repository-services-implementation/src/test/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.LinkedList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.mockito.Mockito.*;
+import static org.testng.Assert.assertEquals;
+
+
+public class OMRSRepositoryContentManagerTest {
```
Comment (mandy-chessell) [inline open-metadata-implementation/repository-services/repository-services-implementation/src/test/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentManagerTest.java:21]: Move brace to next line

File: open-metadata-implementation/repository-services/repository-services-implementation/src/main/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentManager.java [hunk 3]
```diff
@@ -781,6 +781,8 @@ public InstanceType getInstanceType(String          sourceName,
                 instanceType.setTypeDefDescription(typeDef.getDescription());
                 instanceType.setTypeDefDescriptionGUID(typeDef.getDescriptionGUID());
                 instanceType.setTypeDefSuperTypes(this.getSuperTypes(sourceName, typeName, methodName));
+                if (typeDef.getValidInstanceStatusList() != null)
+                    instanceType.setValidStatusList(typeDef.getValidInstanceStatusList());
```
Comment (mandy-chessell) [inline open-metadata-implementation/repository-services/repository-services-implementation/src/main/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentManager.java:785]: Add braces

**Review by mandy-chessell — CHANGES_REQUESTED** [review]
Minor formatting errors - and a question around the use of Mockito

File: open-metadata-implementation/repository-services/repository-services-implementation/src/test/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentManagerTest.java (hunk 1 above)
Comment (keytiong) [reply open-metadata-implementation/repository-services/repository-services-implementation/src/test/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentManagerTest.java]: Mockito was initially used to quickly create a no-op `AuditLog` without cluttering the test case with the chain of`OMRSAuditLogDestination` to `OMRSAuditLogStore`  object instantiation. After your comment, I looked into `AuditLog` implementation and found that the same behaviour can be achieved by passing a null `AuditLogDestination` and removed Mockito.

File: open-metadata-implementation/repository-services/repository-services-implementation/src/test/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentManagerTest.java (hunk 2 above)
Comment (keytiong) [reply open-metadata-implementation/repository-services/repository-services-implementation/src/test/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentManagerTest.java]: Fixed

File: open-metadata-implementation/repository-services/repository-services-implementation/src/main/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentManager.java (hunk 3 above)
Comment (keytiong) [reply open-metadata-implementation/repository-services/repository-services-implementation/src/main/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentManager.java:786]: Fixed

**Review by mandy-chessell — APPROVED** [review]

**Comment by mandy-chessell** [comment]
@keytiong thank you for fixing the test ...

**Pull request merged**